### PR TITLE
remove special condition for workload id for conformance tests

### PR DESCRIPTION
**Pull request by nojnhuh**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:

The [new DRA periodic job](https://prow.k8s.io/view/gs/kubernetes-jenkins/logs/periodic-cluster-api-provider-azure-conformance-with-ci-artifacts-dra-main/1833647506127327232) sets `CONFORMANCE_FLAVOR` in order to use a DRA-enabled template and hits this error in the kind setup script:
```
./scripts/kind-with-registry.sh: line 206: SERVICE_ACCOUNT_SIGNING_PUB: unbound variable
```

The other conformance jobs do not hit the same error because they leave `CONFORMACE_FLAVOR` undefined and let the e2e framework select a default flavor.

IIRC this condition in the kind script to use a different workload ID setup is a vestige of the pre-community infra and/or pre-workload ID default days and no longer is needed now.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

I don't see any other references to `CONFORMANCE_FLAVOR` in test-infra or any other repo, so I don't think this change should include any collateral damage.
https://github.com/search?q=CONFORMANCE_FLAVOR+%28org%3Akubernetes+OR+org%3Akubernetes-sigs%29&type=code

- [ ] cherry-pick candidate

**TODOs**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: scripts/kind-with-registry.sh [hunk 1]
```diff
@@ -203,34 +203,22 @@ function upload_to_blob() {
 # to be mounted on the kind cluster and hence extra mount flags are required.
 function createKindForAZWI() {
   echo "creating workload-identity-enabled kind configuration"
-  if [ -n "${CONFORMANCE_FLAVOR}" ] && [ -n "${SERVICE_ACCOUNT_SIGNING_PUB}" ] && [ -n "${SERVICE_ACCOUNT_SIGNING_KEY}" ]; then
```
Comment (nojnhuh) [inline scripts/kind-with-registry.sh:206]: > IIRC this condition in the kind script to use a different workload ID setup is a vestige of the pre-community infra and/or pre-workload ID default days and no longer is needed now.

@jackfrancis Does that sound right?

Comment (jackfrancis) [reply scripts/kind-with-registry.sh:206]: That sounds plausible, conformance E2E test runs will tell us!

Comment (nojnhuh) [reply scripts/kind-with-registry.sh:206]: Looks like the tests are all far enough along now that they've all created a cluster successfully, so I think we're in the clear.

**Review by mboersma — APPROVED** [review]
/lgtm

**Pull request merged**